script.js

```diff
@@ -107,174 +23,196 @@ document.addEventListener("DOMContentLoaded", function () {
             </div>
         `;
 
-    return actionCard;
-  }
+  return actionCard;
+}
 
-  function generateActionCards() {
-    const actionRow = document.getElementById("actionRow");
-    actions.forEach((action, index) => {
-      // Pass index here
-      const actionCard = generateActionCard(action, index); // Pass both action and index
-      actionRow.appendChild(actionCard);
-    });
-  }
+function generateActionCards() {
+  const actionRow = getElById("actionRow");
+  ACTIONS.forEach((action, index) => {
+    const actionCard = generateActionCard(action, index); // Pass both action and index
+    actionRow.appendChild(actionCard);
+  });
+}
 
-  generateActionCards();
+function getDynamicProperties() {
+  const dynamicProperties = {};
+  const propertyInputs = document.querySelectorAll(
+    "#actionModal input.form-control"
+  );
+  propertyInputs.forEach((input) => {
+    const propertyName = normalizePropertyName(
+      input.getAttribute("data-property-name")
+    );
+    const propertyValue = input.value;
+    dynamicProperties[propertyName] = propertyValue;
+  });
+
+  return dynamicProperties;
+}
+
+// Function to update the live code preview
+function updateLiveCodePreview() {
+  const liveCodeBlock = getElById("liveCodeBlock");
+  const titleElement = getElById("actionModalLabel");
+  const descriptionElement = getElById("actionModalDescription");
+
+  const updatedTrackEvent = {
+    title: titleElement.textContent, // Use textContent instead of value
+    description: descriptionElement.textContent, // Use textContent instead of value
+    ...getDynamicProperties(),
+  };
+
+  const liveCodeContent = liveCodeBlock.querySelector("code");
+  liveCodeContent.textContent = `analytics.track('${
+    updatedTrackEvent.title
+  }', ${JSON.stringify(updatedTrackEvent, null, 2)});`;
+
+  // // Trigger highlight.js syntax highlighting
+  hljs.highlightElement(liveCodeContent);
+  console.log("attempting to highlight using HLS");
+
+  // Update the content of the live code preview
+  liveCodeContent.textContent = `analytics.track('${
+    updatedTrackEvent.title
+  }', ${JSON.stringify(updatedTrackEvent, null, 2)});`;
+}
+
+function addLiveCodePreviewTriggers() {
+  const titleInput = getElById("actionModalLabel");
+  titleInput.addEventListener("input", updateLiveCodePreview);
+
+  const descriptionInput = getElById("actionModalDescription");
+  descriptionInput.addEventListener("input", updateLiveCodePreview);
```

Comment:
the reason why the live code editor wasnt updating when you changed the title/description was because the event listener was `input`, not `change` like with the `propertyInputs`. this is because the title isnt inherantly changable like the `<input>` tag, but made changable with the `contenteditable="true"` attribute on the html